presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/TypeValidator.java

```diff
@@ -140,11 +147,33 @@ public Void visitUnion(UnionNode node, Void context)
             return null;
         }
 
-        private void checkWindowFunctions(Map<Symbol, WindowNode.Function> functions)
+        private void checkWindowFunctions(Map<Symbol, WindowNode.Function> functions, WindowNode.Specification specification)
         {
+            // Inheritance of the following WindowSpecification components is done by the analyzer. As this information
```

Comment:
It's a little weird that this is how the inheritance is done, but I'm not fundamentally opposed to it. I think we should add a couple tests to TestMergeWindows and TestReorderWindows that ensure that nothing screwy happens when we start merging reordering windows that have inherited properties.

Based on this design, I don't expect that that would be a problem, but I think it's worth testing.